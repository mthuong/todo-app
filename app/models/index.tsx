
export * from "./i18n.model";
export * from "./movie.model"
export * from "./item.model"