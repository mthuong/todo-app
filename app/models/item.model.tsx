import { classToClass } from "class-transformer";

export class ItemModel {
  constructor(
    public id: number = 0,
    public title: string = "",
    public selected: boolean = false,
    public isCompleted: boolean = false
  ) { }
}
