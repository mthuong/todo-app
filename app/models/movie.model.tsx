import { classToClass } from "class-transformer";

export class MovieModel {
  constructor(
    public id: number = 0,
    public title: string = "",
    public overview: string = ""
  ) { }
}
