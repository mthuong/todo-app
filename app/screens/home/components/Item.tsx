import * as React from 'react';
import { View, StyleSheet, Text, TouchableOpacity } from 'react-native';
import { ItemModel } from '../../../models/item.model';
import { Row, FlexibleView, IconButton, TextButton } from '../../../components';

interface ItemProps {
  item: ItemModel
  onPressItem: (id: number) => void
  onPressDelete: (id: number) => void
}

export default class Item extends React.Component<ItemProps> {
  _onPress = () => {
    this.props.onPressItem(this.props.item.id);
  };

  _onPressDelete = () => {
    this.props.onPressDelete(this.props.item.id);
  }

  render() {
    const { item } = this.props;
    return (
      <Row style={styles.container}>
        <TouchableOpacity
          onPress={this._onPress}
          style={styles.button}>
          <Text style={[styles.text]}>{this.props.item.title}</Text>
        </TouchableOpacity>
        {item.isCompleted && (<Text>Completed   </Text>)}
        <TextButton text='Delete' onPress={this._onPressDelete} />
      </Row>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    justifyContent: 'center',
    paddingHorizontal: 20,
  },
  button: {
    flex: 1,
    justifyContent: 'center',
  },
  text: {
    flex: 1,
    paddingVertical: 10,
    color: 'black'
  }
})