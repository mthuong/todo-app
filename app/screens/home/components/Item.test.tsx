import React from 'react'
import Enzyme, {shallow} from 'enzyme'
import Item from './Item';
import { ItemModel } from '../../../models';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

describe('Button', () => {
  describe('Rendering', () => {
    it('should match to snapshot', () => {
      const item = new ItemModel(1, 'Title item', false, false)
      const component = shallow(<Item item={item} onPressItem={() => {}} onPressDelete={() => {}} />)
      expect(component).toMatchSnapshot('Item snapshot')
    })
  })
})