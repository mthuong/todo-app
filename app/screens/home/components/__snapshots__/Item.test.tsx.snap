// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Rendering should match to snapshot: Item snapshot 1`] = `ShallowWrapper {}`;
